src/openzaak/components/catalogi/tests/admin/test_import_export_zaaktype.py

```diff
@@ -760,6 +760,112 @@ def test_export_published_zaaktype(self, *mocks):
+        url = reverse("admin:catalogi_zaaktype_change", args=(zaaktype.pk,))
+        response = self.app.get(url)
+        form = response.forms["zaaktype_form"]
+        response = form.submit("_export")
+        data = response.content
+
+        url = reverse("admin:catalogi_catalogus_import_zaaktype", args=(catalogus.pk,))
+        response = self.app.get(url)
+
+        form = response.form
+        f = io.BytesIO(data)
+        f.name = "test.zip"
+        f.seek(0)
+        form["file"] = (
+            "test.zip",
+            f.read(),
+        )
+
+        # one should be new
+        informatieobjecttype2.delete()
+        besluittype2.delete()
+
+        response = form.submit("_import_zaaktype").follow()
+
+        # IOT exists and should be selected
+        iotype_field_0 = response.form["iotype-0-existing"]
+        self.assertEqual(len(iotype_field_0.options), 2)
+        self.assertEqual(  # default option not selected
+            iotype_field_0.options[0], ("", False, _("Create new"))
+        )
+        self.assertEqual(  # option 1 is selected
+            iotype_field_0.options[1],
+            (str(informatieobjecttype.pk), True, str(informatieobjecttype)),
+        )
+        self.assertEqual(iotype_field_0.value, str(informatieobjecttype.pk))
+
+        # IOT does not exist and should select create new
+        iotype_field_1 = response.form["iotype-1-existing"]
+        self.assertEqual(len(iotype_field_1.options), 2)
+        self.assertEqual(  # default option selected
+            iotype_field_1.options[0], ("", True, _("Create new"))
+        )
+        self.assertEqual(  # option 1 not selected
+            iotype_field_1.options[1],
+            (str(informatieobjecttype.pk), False, str(informatieobjecttype)),
+        )
+        self.assertEqual(iotype_field_1.value, "")
+
+        # BesluitType exists and should be selected
+        besluittype_field_0 = response.form["besluittype-0-existing"]
+        self.assertEqual(len(besluittype_field_0.options), 2)
+        self.assertEqual(  # default option not selected
+            besluittype_field_0.options[0], ("", False, _("Create new"))
+        )
+        self.assertEqual(  # option 1 is selected
+            besluittype_field_0.options[1],
+            (str(besluittype1.pk), True, str(besluittype1)),
+        )
+        self.assertEqual(besluittype_field_0.value, str(besluittype1.pk))
+
+        # BesluitType does not exist and should select create new
+        besluittype_field_1 = response.form["besluittype-1-existing"]
+        self.assertEqual(len(besluittype_field_1.options), 2)
+        self.assertEqual(  # default option selected
+            besluittype_field_1.options[0], ("", True, _("Create new"))
+        )
+        self.assertEqual(  # option 1 not selected
+            besluittype_field_1.options[1],
+            (str(besluittype1.pk), False, str(besluittype1)),
+        )
+        self.assertEqual(besluittype_field_1.value, "")
```

Comment:
Nice test! Please add also submitting this form with default choices and assertion that the new zaaktypen have relations with existing pre-selected iotypen

Reply:
Added a test for this.